enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx

```diff
@@ -72,41 +72,71 @@ export const SdkIframeEmbedSetupProvider = ({
     ...(settings.questionId && { questionId: Number(settings.questionId) }),
   });
 
-  const updateSettings = useCallback(
-    (nextSettings: Partial<SdkIframeEmbedSettings>) =>
-      setSettings(
-        (prevSettings) =>
-          ({
-            ...prevSettings,
-            ...nextSettings,
-          }) as SdkIframeEmbedSettings,
-      ),
-    [setSettings],
+  const fallbackDashboardId =
```

Comment:
There could be a race condition where `recentDashboards` are not yet fully loaded, yet `handleEmbedSettingsRestored` are already called by the `usePersistJsonViaUserSetting` hook. I'll have to look into that.